app/code/Magento/GraphQl/Model/Query/ContextDataProvider.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\GraphQl\Model\Query;
+
+use Magento\Framework\Api\ExtensibleDataObjectConverter;
+use Magento\Framework\App\ObjectManager;
+
+class ContextDataProvider implements ContextDataProviderInterface
+{
+    /** @var ContextInterface */
+    private $context;
+
+    /** @var ContextFactoryInterface */
+    private $contextFactory;
+
+    /** @var ExtensibleDataObjectConverter  */
+    private $dataObjectConverter;
+
+    /**
+     * ContextDataProvider constructor.
+     * @param ContextFactoryInterface $contextFactory
+     * @param ExtensibleDataObjectConverter $dataObjectConverter
+     */
+    public function __construct(
+        ContextFactoryInterface $contextFactory,
+        ExtensibleDataObjectConverter $dataObjectConverter
+    ) {
+        $this->contextFactory = $contextFactory ?: ObjectManager::getInstance()->get(ContextFactoryInterface::class);
```

Comment:
Need not validate null params for **new** classes as it wont break any previous installations